packages/compass-web/sandbox/sandbox-atlas-preferences.tsx

```diff
@@ -0,0 +1,84 @@
+import { useEffect, useState } from 'react';
+
+type AtlasPreferences = {
+  enableGenAIFeaturesAtlasProject: boolean;
+  enableGenAISampleDocumentPassingOnAtlasProject: boolean;
+  enableGenAIFeaturesAtlasOrg: boolean;
+  optInDataExplorerGenAIFeatures: boolean;
+};
+
+type AtlasPreferencesReturnValue =
+  | {
+      status: 'loading';
+      preferences: null;
+    }
+  | { status: 'loaded'; preferences: AtlasPreferences };
+
+export function useAtlasPreferences({
+  projectId,
+}: {
+  projectId?: string;
+}): AtlasPreferencesReturnValue {
+  const [preferences, setAtlasPreferences] = useState<AtlasPreferences | null>(
+    null
+  );
+
+  const overrideGenAIEnablement =
+    process.env.COMPASS_WEB_GEN_AI_ENABLEMENT === 'true';
+
+  useEffect(() => {
+    if (!projectId || overrideGenAIEnablement) {
+      return;
+    }
+
+    const fetchPreferences = async () => {
+      const {
+        appUser: { isOptedIntoDataExplorerGenAIFeatures },
+        currentOrganization: { genAIFeaturesEnabled },
+        featureFlags: { groupEnabledFeatureFlags },
+      } = await fetch(`/cloud-mongodb-com/v2/${projectId}/params`).then(
```

Comment:
We already fetch those in sandbox sign in hook, you can just return these values from there